File: android/word/src/main/java/com/dreampany/word/ui/activity/WebActivity.java
/*
package com.dreampany.word.ui.activity;

import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Bundle;

import com.dreampany.frame.databinding.ActivityWebBinding;
import com.dreampany.frame.ui.activity.BaseActivity;
import com.dreampany.word.R;
import com.dreampany.word.ui.model.UiTask;

import org.jetbrains.annotations.Nullable;

import im.delight.android.webview.AdvancedWebView;

*/
/**
 * Created by Roman on 1/23/2019
 * Copyright (c) 2019 Dreampany. All rights reserved.
 * dreampanymail@gmail.com
 * Last modified $file.lastModified
 *//*

public class WebActivity extends BaseActivity implements AdvancedWebView.Listener {

    ActivityWebBinding binding;

    @Override
    public int getLayoutId() {
        return R.layout.activity_web;
    }

    @Override
    public boolean isFullScreen() {
        return true;
    }

    @Override
    protected void onStartUi(@Nullable Bundle state) {
        binding = (ActivityWebBinding) super.binding;
        binding.webview.setListener(this, this);

        UiTask<?> task = getCurrentTask(false);
        if (task == null) {
            finish();
            return;
        }
        String url = task.getComment();
        binding.webview.loadUrl(url);
    }

    @Override
    protected void onStopUi() {
        binding.webview.onDestroy();
    }

    @Override
    protected void onResume() {
        super.onResume();
        binding.webview.onResume();
    }

    @Override
    protected void onPause() {
        binding.webview.onPause();
        super.onPause();
    }

    @Override
    public void onBackPressed() {
        if (!binding.webview.onBackPressed()) { return; }
        super.onBackPressed();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
        super.onActivityResult(requestCode, resultCode, intent);
        binding.webview.onActivityResult(requestCode, resultCode, intent);
    }

    @Override
    public void onPageStarted(String url, Bitmap favicon) {

    }

    @Override
    public void onPageFinished(String url) {

    }

    @Override
    public void onPageError(int errorCode, String description, String failingUrl) {

    }

    @Override
    public void onDownloadRequested(String url, String suggestedFilename, String mimeType, long contentLength, String contentDisposition, String userAgent) {

    }

    @Override
    public void onExternalPageRequest(String url) {

    }
}
*/

File: android/framework_old/src/main/kotlin/com/dreampany/framework/ui/activity/WebActivity.kt
package com.dreampany.framework.ui.activity

import android.content.Intent
import android.graphics.Bitmap
import android.os.Bundle
import android.view.View
import com.dreampany.framework.R
import com.dreampany.framework.data.model.Task
import com.dreampany.framework.databinding.ActivityWebBinding
import im.delight.android.webview.AdvancedWebView

/**
 * Created by roman on 2019-07-27
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class WebActivity : BaseActivity(), AdvancedWebView.Listener {

    private lateinit var bind: ActivityWebBinding

    override fun getLayoutId(): Int {
        return R.layout.activity_web
    }

    override fun isFullScreen(): Boolean {
        return true
    }

    override fun onStartUi(state: Bundle?) {
        bind = super.binding as ActivityWebBinding

        val task = getCurrentTask<Task<*>>(false)
        if (task == null) {
            finish()
            return
        }
        bind.webView.setListener(this, this)
        val url = task.extra
        bind.webView.loadUrl(url)
    }

    override fun onStopUi() {
        bind.webView.onDestroy()
    }

    override fun onResume() {
        super.onResume()
        bind.webView.onResume()
    }

    override fun onPause() {
        bind.webView.onPause()
        super.onPause()
    }

    override fun onBackPressed() {
        if (!bind.webView.onBackPressed()) {
            return
        }
        super.onBackPressed()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
        super.onActivityResult(requestCode, resultCode, intent)
        bind.webView.onActivityResult(requestCode, resultCode, intent)
    }

    override fun onPageStarted(url: String, favicon: Bitmap?) {
        bind.viewSpin.visibility = View.VISIBLE
    }

    override fun onPageFinished(url: String) {
        bind.viewSpin.visibility = View.GONE
    }

    override fun onPageError(errorCode: Int, description: String?, failingUrl: String?) {
        bind.viewSpin.visibility = View.GONE
    }

    override fun onDownloadRequested(
        url: String,
        suggestedFilename: String?,
        mimeType: String?,
        contentLength: Long,
        contentDisposition: String?,
        userAgent: String?
    ) {

    }

    override fun onExternalPageRequest(url: String) {
    }
}